src/Uno.UI/UI/Xaml/Automation/Peers/FrameworkElementAutomationPeer.cs

```diff
@@ -18,200 +27,575 @@
+					var apAnnotation = new AutomationPeerAnnotation();
+
+					// Retrieve information from the AutomationAnnotation
+					var type = uiElementAnnotation.Type;
+					var uie = uiElementAnnotation.Element;
+
+					// Get or create AutomationPeer for the UIElement
+					AutomationPeer ap = null;
+					if (uie != null)
+					{
+						ap = uie.GetOrCreateAutomationPeer();
+					}
+
+					// Set properties of AutomationPeerAnnotation
+					apAnnotation.Type = type;
+					apAnnotation.Peer = ap;
+
+					annotations.Add(apAnnotation);
+				}
 			}
 		}
 
-		protected override bool IsEnabledCore()
+		return annotations;
+	}
+
+	public void SetControlType(AutomationControlType controlType)
+	{
+		m_ControlType = controlType;
+	}
+
+	public void SetLocalizedControlType(string localizedControlType)
+	{
+		m_LocalizedControlType = localizedControlType;
+	}
+
+	public void SetClassName(string className)
+	{
+		m_ClassName = className;
+	}
+
+	public static AutomationPeer FromElement(UIElement element)
+	{
+		if (element is IFrameworkElement fe)
+		{
+			return FromIFrameworkElement(fe);
+		}
+
+		return null;
+	}
+
+	public static AutomationPeer CreatePeerForElement(UIElement element)
+	{
+		if (element is IFrameworkElement fe)
 		{
-			if (Owner is Control control)
+			return CreatePeerForIFrameworkElement(fe);
+		}
+
+		return null;
+	}
+
+	private static AutomationPeer CreatePeerForIFrameworkElement(IFrameworkElement element)
+	{
+		if (element is not { })
+		{
+			throw new ArgumentNullException(nameof(element));
+		}
+
+		return element.GetAutomationPeer();
+	}
+
+	private static AutomationPeer FromIFrameworkElement(IFrameworkElement element)
+	{
+		if (element is not { })
+		{
+			throw new ArgumentNullException(nameof(element));
+		}
+
+		return element.GetAutomationPeer();
+	}
+
+	/// <summary>
+	/// Virtual helper method which provide ability for any specific Automation peers
+	/// do not allows including Automation peer of child elements' in to the Automation
+	/// peer tree.
+	/// </summary>
+	/// <remarks>
+	/// We don't accept nonUI or null elements by default.
+	/// </remarks>
+	/// <param name="child">
+	/// Child element to be decided to include it to
+	/// Automation peer's tree.
+	/// </param>
+	/// <returns>True if the child element is acceptable.</returns>
+	private protected virtual bool ChildIsAcceptable(UIElement element)
+	{
+		var childIsAcceptable = element != null;
+
+		if (element is { })
+		{
+			var isPopupOpen = true;
+
+			if (element is Popup popup)
 			{
-				return control.IsEnabled;
+				isPopupOpen = popup.IsOpen;
 			}
+			var value = element.Visibility;
 
-			return true;
+			// this condition checks that if Control is visible and if it's popup then it must be open
+			childIsAcceptable = isPopupOpen && value == Visibility.Visible;
 		}
 
-		protected override void SetFocusCore()
+		return childIsAcceptable;
+	}
+
+	public IReadOnlyList<AutomationPeer> GetAutomationPeerCollection(DependencyProperty eProperty)
```

Comment:
Make sure only the public members from WinUI are also public on our side -  
![image](https://github.com/unoplatform/uno/assets/1075116/d2a0d5e7-9cd1-4239-914d-2a6946b58ff1)